common/client.go

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/kinesis"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+func NewClient(ctx context.Context, httpClient *http.Client, cfg Config) (*kinesis.Client, error) {
```

Comment:
I think we should add a comment for this exported method. Even it could look self explanatory, it'd be nice to have for the generated go documentation.

Reply:
You are right, and I should actually document why I'm passing a custom httpClient, doesn't make sense without looking at the internals.